src/CMakeLists.txt

```diff
@@ -30,11 +30,20 @@ mark_as_advanced(EXECUTABLE_OUTPUT_PATH LIBRARY_OUTPUT_PATH)
 option(ENABLE_CURL "Enable cURL support for fetching media" TRUE)
 set(USE_CURL FALSE)
 
+if(NOT BUILD_CLIENT)
+	set(BUILD_SERVER TRUE)
+	set(ENABLE_CURL TRUE)
+endif()
+
 if(ENABLE_CURL)
 	find_package(CURL)
 	if (CURL_FOUND)
 		message(STATUS "cURL support enabled.")
 		set(USE_CURL TRUE)
+	else(BUILD_SERVER)
+		message(WARNING "cURL is required to announce to the server list")
+	else(BUILD_CLIENT)
```

Comment:
`BUILD_CLIENT` and `BUILD_SERVER` are not exclusive. If both are set, both warning messages should be displayed.